public/app/plugins/panel/geomap/utils/location.ts

```diff
@@ -199,3 +275,16 @@ function getPointsFromGeohash(field: Field<string>): Point[] {
   }
   return points;
 }
+
+function getPointsFromLookup(field: Field<string>, locationData: any): Point[] {
+  const count = field.values.length;
+  const points = new Array<Point>(count);
+  for (let i = 0; i < count; i++) {
+    const target = field.values.get(i);
+    const location = locationData.filter((loc: any) => loc.key.toUpperCase() === target.toUpperCase())[0];
```

Comment:
This approach has you calling `toUpperCase` for on every location for each value.

Consider preparing the original location data as a Map<string,...> where the keys are already uppercase.